_user/voxeldemo/main.c

```diff
@@ -81,14 +84,19 @@ int voxel_init(voxel_t *v)
 		if ((v->mapColor = malloc(1024 * 1024)) == NULL)
 			break;
 
-		if ((v->bufBack = malloc(v->width * sizeof(int))) == NULL)
+		if ((v->bufBack = malloc(g->width * sizeof(int))) == NULL)
 			break;
 
-		if ((v->palette = malloc(256 * sizeof(uint32_t))) == NULL)
+		if ((v->pixels = malloc(g->width * g->height * g->depth)) == NULL)
 			break;
 
+		if ((v->palette = malloc(256 * g->depth)) == NULL)
+			break;
+
+		v->width = g->width;
+		v->height = g->height;
 
-		return 0;
+		return EOK;
 	} while (0);
 
 	voxel_free(v);
```

Comment:
What value will have the `v->mapcolor variable`
if  `(v->mapHeight = malloc(1024 * 1024)) == NULL` will be true?
You should set variables before allocating memory.
```
v->mapColor = NULL;
v->bufback = NULL;
v->pixels = NULL;
v->palette = NULL;
```

Reply:
It is expected to set `voxel_t` structure/context properly before `voxel_init(..)` is called, thus:
```
	voxel_t v = {
		.cam = { .h = 300, .horiz = 200, .dist = 300 },
	};
```
is enough, any other member of the `voxel_t` structure is/are set to `NULL`, `0` by default, so all `mapHeight`, `mapColor`, `backBuf`, `pixels` and `palette` are set to `NULL` by default, (this is property of C initializer), before call to `voxel_init(..)`.

So, to answer Your question `mapHeight` will be `NULL`, but `mapColor` will be assigned a valid pointer returned by successful `malloc`, later `voxel_free(v)` is called and function returns an error, program fail then, and exits.